integration-tests/gradle/build.gradle.kts

```diff
@@ -22,6 +22,8 @@ tasks.integrationTest {
 
     javaLauncher.set(javaToolchains.launcherFor {
         // kotlinx.coroutines requires Java 11+
-        languageVersion.set(JavaLanguageVersion.of(11))
+        languageVersion.set(dokkaBuild.testJavaLauncherVersion.map {
+            maxOf(it, JavaLanguageVersion.of(11))
+        })
     })
 }
```

Comment:
Following on from https://github.com/Kotlin/dokka/pull/2918#issuecomment-1471193510 @IgnatBeresnev 

> Non-blocking nitpick, will merge without it, but do let me know if you want to address it in this PR: maybe [the fix](https://github.com/Kotlin/dokka/blob/3ea414251d32108dab8b14d41a672e9834824690/integration-tests/gradle/build.gradle.kts#L23) we put in for the integration tests could be revisited? Is there a better option after current changes, like setting the new build property value to `11` and letting the convention plugin do the configuration part? GitHub actions configuration for them could be corrected as well it seems
> 
> > Update TeamCity to use the new variables (if necessary?)
> 
> Doesn't look like it needs to be updated - they only run the `test` task, and have the same JVM version as the default one, so it should be good. The matrix thing is GitHub Actions only

Good hint! I had an idea, so hopefully at least Java 11 and 17 will be integration tested.

Unfortunately because all the integration tests are lumped together, they need to use the same Java launcher, which isn't ideal.

Potential solutions

* update the Coroutines build config to support toolchains https://github.com/Kotlin/kotlinx.coroutines/issues/3665
* add unit-test tags for each test project, and create separate Gradle test tasks for each tag, so the launcher can be specified per project
* use JUnit conditionals to only run KxS on Java 11+ (JUnit 5 docs: https://junit.org/junit5/docs/current/user-guide/#writing-tests-conditional-execution-jre)

Reply:
Setting the min version to 11 with the possibility to go above is good enough for now, I think :+1: 

We can keep this info in mind for when the integration test runner is refactored, maybe by that time kotlinx.coroutines will support toolchains or the integration tests themselves won't be as coupled

I should probably create an issue for it to not lose context, and quote your message and proposals